### ISSUE TITLE
Page Jump, Anchor Links, Jump Links

### ISSUE THREAD

**Issue by geoffgolder-ah** [contentful/rich-text#124]
I am trying to have a link to go to a certain section of the page, this basically means setting the `name` field of an <a> tag and linking to it.  Obviously I can link to it, but I am not able to actually set the name of a link.

Is there a way to do this I am missing, or a way to override the actual rich text editor so that it allows us to set the "name" attribute on the link tag?

**Comment by AndreiEtumian-earnin** [comment]
@geoffgolder-ah Did you find a solution in the end? I am having the same issue right now.

**Comment by geoffgolder-ah** [comment]
No.  Effective duplicate of this: https://github.com/contentful/rich-text/issues/56

**Comment by parkersweb** [comment]
#56 has been closed - am I right in thinking that the official answer is "fork our editor and write your own"?

**Comment by jeniamoroz** [comment]
> #56 has been closed - am I right in thinking that the official answer is "fork our editor and write your own"?

"and then we will adopt your solution", must admit not a bad move :)

**Comment by konstantib** [comment]
I'm struggling to explain to my Product owner why such basic and commonly used feature is not available in Rich Text editor.
One solution that everyone seems to be suggesting is to create Content Type to be used for Headings (or any other text to be jump/anchor linked to) and then use RTE hyperlink functionality to link to entry when creating a link itself. This should be enough data to render Jump links on the FE. However creating yet another content type just for this seems like an overkill.

**Comment by sb8244** [comment]
This feels like a non-starter for Contentful :/ I really want to recommend it but I cannot understand why the rich text editing framework at a company focused around content won't prioritize these types of features.

**Comment by waltercolindres** [comment]
Hi, curious if this is on a roadmap? 2+ years, I am assuming this will be closed?

**Comment by sebbean** [comment]
how we doin?

**Comment by EliseiNicolae** [comment]
Any updates on this one?

**Comment by pavelrahman619** [comment]
any updates?

**Comment by norahine** [comment]
Same, any updates on this?

**Comment by hapiben** [comment]
Any updates?

**Comment by Lucsy3012** [comment]
We could really need this in November 2022 as well. Updates here would be great

**Comment by ShaunMWallace** [comment]
Adding another nice to have here

**Comment by bdogaru** [comment]
I guess still nothing yet?

Not the same but an alternative would be building a Table of Contents component out of the headings. Live example [here](https://vevolmedia.com/blog/plugins-and-apps-for-shopify-stores)

**Comment by yoshyosh** [comment]
Hi all, I highly recommend using chatgpt to help with this. You can do it with javascript to grab all the headings and dynamically create anchor links. A simple example to start:

```
// Iterate over the <h2> elements
h2Elements.forEach(function(h2, index) {
    const liElement = document.createElement('li');
    const aElement = document.createElement('a');
    const anchorId = 'toc_' + index; // Using string concatenation
    h2.id = anchorId; // Set an id for the <h2>
    aElement.href = '#' + anchorId; // Using string concatenation
    aElement.textContent = h2.textContent;
    
    liElement.appendChild(aElement);
    olElement.appendChild(liElement);
});
```

But I think it's likely best to parse the contentful entry with nokogiri first to add to the page, then use javascript to add the anchor text to each heading. This is also helpful for SEO.

**Comment by jtsierra** [comment]
Any updates on this? Might be handy to have anchor links in 2024.
